apps/stats/tests/test_views.py

```diff
@@ -878,3 +882,104 @@ def test_report_view_xss(self):
+                'slug': 'a3615', 'year': '2016', 'month': '01'}))
+
+        assert response.status_code == 401
+
+    def test_list_does_not_own_addon(self):
+        self.user = UserProfile.objects.create(
+            read_dev_agreement=datetime.datetime.now())
+        self.api_key = self.create_api_key(self.user, 'bar')
+        response = self.get(
+            reverse('stats.archive_list', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01'}))
+        assert response.status_code == 403
+
+    def test_list(self):
+        self.user = UserProfile.objects.create(
+            read_dev_agreement=datetime.datetime.now())
+        self.api_key = self.create_api_key(self.user, 'bar')
+        AddonUser.objects.create(user=self.user, addon=self.addon)
+
+        save_stats_to_file(self.theme_update_count)
+
+        response = self.get(
+            reverse('stats.archive_list', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01'}))
+        assert response.status_code == 200
+        assert json.loads(response.content) == [
+            {
+                'date': '2016-01-18',
+                'addon_id': 3615,
+                'model_name': 'themeupdatecount',
+            }
+        ]
+
+    def test_list_not_existing(self):
+        self.user = UserProfile.objects.create(
+            read_dev_agreement=datetime.datetime.now())
+        self.api_key = self.create_api_key(self.user, 'bar')
+        AddonUser.objects.create(user=self.user, addon=self.addon)
+
+        response = self.get(
+            reverse('stats.archive_list', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01'}))
+        assert response.status_code == 404
+
+    def test_get(self):
+        self.user = UserProfile.objects.create(
+            read_dev_agreement=datetime.datetime.now())
+        self.api_key = self.create_api_key(self.user, 'bar')
+        AddonUser.objects.create(user=self.user, addon=self.addon)
+
+        save_stats_to_file(self.theme_update_count)
+
+        response = self.get(
+            reverse('stats.archive', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01',
+                'day': '18', 'model_name': 'themeupdatecount'}))
+        assert response.status_code == 200
+        assert json.loads(response.content) == {
+            'count': 123,
+            'date': '2016-01-18',
+            'addon': 3615
+        }
+
+    def test_get_not_existing(self):
+        self.user = UserProfile.objects.create(
+            read_dev_agreement=datetime.datetime.now())
+        self.api_key = self.create_api_key(self.user, 'bar')
+        AddonUser.objects.create(user=self.user, addon=self.addon)
+
+        response = self.get(
+            reverse('stats.archive', kwargs={
+                'slug': 'a3615', 'year': '2016', 'month': '01',
+                'day': '18', 'model_name': 'themeupdatecount'}))
+        assert response.status_code == 404
```

Comment:
Not sure I understand what you mean, this tests if the data exists. The addon is owned by the user and the user is able to use the api auth, so I'm testing the specific 404-part.